pkg/volume/iscsi/iscsi.go

```diff
@@ -116,40 +127,52 @@ func (plugin *iscsiPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volume.V
 func (plugin *iscsiPlugin) newMounterInternal(spec *volume.Spec, podUID types.UID, manager diskManager, mounter mount.Interface, exec mount.Exec, secret map[string]string) (volume.Mounter, error) {
 	// iscsi volumes used directly in a pod have a ReadOnly flag set by the pod author.
 	// iscsi volumes used as a PersistentVolume gets the ReadOnly flag indirectly through the persistent-claim volume used to mount the PV
-	iscsi, readOnly, err := getVolumeSource(spec)
+	readOnly, fsType, err := getISCSIVolumeInfo(spec)
+	if err != nil {
+		return nil, err
+	}
+	tp, portals, iqn, lunStr, err := getISCSITargetInfo(spec)
 	if err != nil {
 		return nil, err
 	}
 
-	lun := strconv.Itoa(int(iscsi.Lun))
-	portal := portalMounter(iscsi.TargetPortal)
+	lun := strconv.Itoa(int(lunStr))
+	portal := portalMounter(tp)
 	var bkportal []string
 	bkportal = append(bkportal, portal)
-	for _, tp := range iscsi.Portals {
+	for _, tp := range portals {
```

Comment:
"tp" is used just few line above for other purpose. Can you use another parameter name to avoid confusion?